### DISCUSSION TITLE
How do I configure for ARM64-windows?

### DISCUSSION THREAD

**Discussion by mijagourlay** [microsoft/vcpkg#22923] [Q&A]
I am trying to acquire or build libcurl for ARM64 on Windows (specifically for HoloLens 2), using vcpkg.

(I understand that vcpkg is more general than libcurl or zlib but I'm describing my ultimate goal for context.)

I tried "vcpkg.exe install curl[tool]:arm64-windows" but that fails (apparently due to zlib not supporting that triplet):

vcpkg>vcpkg.exe install curl[tool]:arm64-windows
Computing installation plan...
The following packages will be built and installed:
    curl[core,non-http,schannel,ssl,sspi,tool]:arm64-windows -> 7.81.0
  * zlib[core]:arm64-windows -> 1.2.11#13
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet arm64-windows...
Error: in triplet arm64-windows: Unable to find a valid toolchain combination.
    The requested target architecture was arm64
    The selected Visual Studio instance is at C:\Program Files\Microsoft Visual Studio\2022\Enterprise
    The available toolchain combinations are x86, amd64, x86_amd64, amd64_x86

See https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#VCPKG_VISUAL_STUDIO_PATH for more information.
Note: Updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.

Regarding the above Note: This was just after running "bootstrap-vcpkg".

vcpkg reports that arm64-windows is one of the built-in triplets, so I would have expected the above to work.

I saw another discussion about using vcpkg to configure zlib but the symptom was not the same as the one above.
(https://github.com/microsoft/vcpkg/issues/15823)

How does one build or obtain libcurl for arm64-windows?

Thanks!

Below is the output of "vcpkg.exe install zlib --triplet=arm64-windows --debug":

vcpkg>vcpkg.exe install zlib --triplet=arm64-windows --debug
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Failed to open: C:\Users\mijag\GitHub\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=0, usegitregistry=0, embeddedsha=nullopt
[DEBUG] Using builtin-ports: C:\Users\mijag\GitHub\vcpkg\ports
[DEBUG] Using installed-root: C:\Users\mijag\GitHub\vcpkg\installed
[DEBUG] Using buildtrees-root: C:\Users\mijag\GitHub\vcpkg\buildtrees
[DEBUG] Using packages-root: C:\Users\mijag\GitHub\vcpkg\packages
[DEBUG] Using scripts-root: C:\Users\mijag\GitHub\vcpkg\scripts
[DEBUG] Using vcpkg-root: C:\Users\mijag\GitHub\vcpkg
[DEBUG] Using scripts-root: C:\Users\mijag\GitHub\vcpkg\scripts
[DEBUG] Using builtin-registry: C:\Users\mijag\GitHub\vcpkg\versions
[DEBUG] Using downloads-root: C:\Users\mijag\GitHub\vcpkg\downloads
[DEBUG] Default binary cache path is: C:\Users\mijag\AppData\Local\vcpkg\archives
Computing installation plan...
[DEBUG] Found path: C:\Program Files\CMake\bin\cmake.exe
[DEBUG] CreateProcessW("C:\Users\mijag\GitHub\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe" --version)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 11032: cmd_execute_and_stream_data() returned 0 after    77080 us
[DEBUG] CreateProcessW("C:\Users\mijag\GitHub\vcpkg\downloads\tools\cmake-3.21.1-windows\cmake-3.21.1-windows-i386\bin\cmake.exe" -DVCPKG_ROOT_DIR=C:/Users/mijag/GitHub/vcpkg -DPACKAGES_DIR=C:/Users/mijag/GitHub/vcpkg/packages -DBUILDTREES_DIR=C:/Users/mijag/GitHub/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=C:/Users/mijag/GitHub/vcpkg/installed -DDOWNLOADS=C:/Users/mijag/GitHub/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "C:\Users\mijag\GitHub\vcpkg\buildtrees\0.vcpkg_tags.cmake")
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 11032: cmd_execute_and_stream_data() returned 0 after    42413 us
The following packages will be built and installed:
    zlib[core]:arm64-windows -> 1.2.11#13
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy -products * -format xml)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 11032: cmd_execute_and_stream_data() returned 0 after    56751 us
Detecting compiler hash for triplet arm64-windows...
[DEBUG] Found path: C:\Program Files\Git\cmd\git.exe
[DEBUG] CreateProcessW("C:\Program Files\Git\cmd\git.exe" --version)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 11032: cmd_execute_and_stream_data() returned 0 after    45989 us
Error: in triplet arm64-windows: Unable to find a valid toolchain combination.
    The requested target architecture was arm64
    The selected Visual Studio instance is at C:\Program Files\Microsoft Visual Studio\2022\Enterprise
    The available toolchain combinations are x86, amd64, x86_amd64, amd64_x86

See https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#VCPKG_VISUAL_STUDIO_PATH for more information.
Note: Updating vcpkg by rerunning bootstrap-vcpkg may resolve this failure.
[DEBUG] D:\a\_work\1\s\src\vcpkg\build.cpp(357)
[DEBUG] Time in subprocesses: 222233 us
[DEBUG] Time in parsing JSON: 276 us
[DEBUG] Time in JSON reader: 251 us
[DEBUG] Time in filesystem: 3996 us
[DEBUG] Time in loading ports: 2065 us
[DEBUG] Exiting after 284.5 ms (271908 us)

**Comment by Osyotr** [comment]
> The available toolchain combinations are x86, amd64, x86_amd64, amd64_x86

Do you have arm component installed in visual studio?
